src/main/java/decodes/tsdb/algo/AverageAlgorithm.java

```diff
@@ -1,84 +1,71 @@
 package decodes.tsdb.algo;
 
-import java.util.Date;
-
-import ilex.var.NamedVariableList;
 import ilex.var.NamedVariable;
-import decodes.tsdb.DbAlgorithmExecutive;
 import decodes.tsdb.DbCompException;
-import decodes.tsdb.DbIoException;
-import decodes.tsdb.VarFlags;
 import decodes.util.PropertySpec;
+import org.opendcs.annotations.algorithm.Algorithm;
+import org.opendcs.annotations.algorithm.Input;
+import org.opendcs.annotations.algorithm.Output;
 
-//AW:JAVADOC
-/**
-AverageAlgorithm averages single 'input' parameter to a single 'average' 
-parameter. The averaging period is determined by the interval of the output
-parameter.
+@Algorithm(
+		description = "AverageAlgorithm averages single 'input' parameter to a single 'average' \n" +
+				"parameter. The averaging period is determined by the interval of the output\n" +
+				"parameter."
+)
 
- */
-//AW:JAVADOC_END
 public class AverageAlgorithm extends decodes.tsdb.algo.AW_AlgorithmBase
 {
-//AW:INPUTS
-	double input;	//AW:TYPECODE=i
-	String _inputNames[] = { "input" };
-//AW:INPUTS_END
+	private static final String AVERAGESTRING = "average";
+	@Input
+	double input;
 
-//AW:LOCALVARS
 	double tally;
 	int count;
-	PropertySpec specs[] = 
+	PropertySpec specs[] =
```

Comment:
you can delete this list and the associated method. The base class now takes care of it given the annotation below.